src/Cake.NuGet/Install/NuGetPackageInstaller.cs

```diff
@@ -115,6 +131,22 @@ public IReadOnlyCollection<IFile> Install(PackageReference package, PackageType
             return project.GetFiles(path, package, type);
         }
 
+        private static DirectoryPath GetPackagePath(IDirectory root, string package)
+        {
+            var directories = root.GetDirectories("*", SearchScope.Current).ToArray();
+            return directories.FirstOrDefault(p => p.Path.GetDirectoryName().Equals(package, StringComparison.OrdinalIgnoreCase))?.Path;
+        }
+
+        private static DirectoryPath GetPackagePath(DirectoryPath root, PackageReference package)
+        {
+            if (package.Parameters.ContainsKey("version"))
+            {
+                var version = package.Parameters["version"].First();
+                return root.Combine($"{package.Package}.{version}".ToLowerInvariant());
+            }
+            return root.Combine(package.Package.ToLowerInvariant());
```

Comment:
In-Process NuGet client will always have version appended to directory name, even if version is not specified in the directive.